sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/WebExceptionUtility.java

```diff
@@ -76,7 +77,8 @@ private static boolean isNetworkFailureInternal(Exception ex) {
         if (ex instanceof ClosedChannelException
             || ex instanceof SocketException
             || ex instanceof SSLException
-            || ex instanceof UnknownHostException) {
+            || ex instanceof UnknownHostException
+            || ex instanceof PrematureCloseException) {
```

Comment:
should it also be added in isWebExceptionRetriableInternal?

Reply:
No - these exceptions all indicate that the request never made it to the service - so it is safe to retry even for writes (see the comment above the if statement). PrmeatureCloseException means a TCP reset was received before receiving the response - but it is unknown whether the request made it to the service endpoint.

Reply:
gotcha, thanks

Reply:
I think this will introduced the regression where on PrmeatureCloseException  sdk will do the failover. I will discuss it offline and we might need to revert the pr